hw/ip/i2c/data/i2c_testplan.hjson

```diff
@@ -608,29 +613,83 @@
-              - If the transaction ends due to this timeout, the STATUS.HOST_DISABLED_NACK_TIMEOUT
-                bit is set. This bit is cleared upon starting a new transaction.
+                end the transaction, sending a Stop condition and transitioning the bus to Idle.
+              - If the transaction ends due to this timeout, the CONTROLLER_EVENTS.UNHANDLED_NACK_TIMEOUT
+                bit is set to alert software.
 
             Stimulus
               - Configure DUT/Agent to Host/Target mode respectively.
               - Stimulate transactions, and randomly allow the Target to NACK.
-              - Randomize the HOST_NACK_HANDLER_TIMEOUT / 'nak' handler latency, to allow the timeout
-                behaviour to trigger.
+              - Randomize the HOST_NACK_HANDLER_TIMEOUT / 'controller_halt' handler latency, to allow
+                the timeout behaviour to trigger.
               - Software should randomly disable the HOST_NACK_HANDLER_TIMEOUT mid-stretch, to disable
                 the automatic stop generation.
 
             Checking:
-              - Data validity continues to match expectation when software uses the 'nak' interrupt
-                to change up the dataflow.
-              - During halt, the bus should be idle (both signals released)
-              - If the 'nak' interrupt is unhandled, after the HOST_NACK_HANDLER_TIMEOUT delay the DUT
-                should be observed to have returned to disabled, and a Stop condition should have been
-                generated to end the active transaction.
-              - Check that the STATUS.HOST_DISABLED_NACK_TIMEOUT bit is set when the timeout is reached.
+              - Data validity continues to match expectation when software uses the 'controller_halt'
+                interrupt to change up the dataflow.
+              - During halt, the bus should be have SCL asserted, SDA released.
+              - If the 'controller_halt' interrupt is unhandled, after the HOST_NACK_HANDLER_TIMEOUT delay
+                the Controller FSM should be observed returning to Idle, and a Stop condition should have
+                been generated to end the active transaction.
+              - Check that the CONTROLLER_EVENTS.UNHANDLED_NACK_TIMEOUT bit is set when the timeout is
+                reached.
+            '''
+      stage: V2
+      tests: []
+    }
+    {
+      name: target_mode_n_byte_ack_control
+      desc: '''
+            Test the (Target-Mode) N-Byte ACK control mode
+            When this mode is enabled, we should expect:
+            - So long as there is space in the ACQ_FIFO, write data bytes will be ACK'd if the
+              auto-decremeting ACK Counter (TARGET_ACK_CTRL.NBYTES) contains a positive value.
+            - If the Auto ACK Counter is zero, the clock is stretched up to the timeout configured in
+              TARGET_TIMEOUT_CTRL. If the timeout is reached (or is configured to zero/disabled), the
```

Comment:
Something like this would be more accurate:

```suggestion
              TARGET_TIMEOUT_CTRL, if the timeout is enabled. If the timeout is reached (or is configured to zero), the
```

If TARGET_TIMEOUT_CTRL is disabled, the target FSM waits / stretches indefinitely.